sdk/storage/Azure.Storage.Blobs/src/PartitionedDownloader.cs

```diff
@@ -30,20 +31,40 @@ internal class PartitionedDownloader
 
         public PartitionedDownloader(
             BlobBaseClient client,
-            StorageTransferOptions transferOptions = default,
-            long? initialTransferLength = null)
+            StorageTransferOptions transferOptions = default)
         {
             _client = client;
-            _maxWorkerCount =
-                transferOptions.MaximumConcurrency ??
-                Constants.Blob.Block.DefaultConcurrentTransfersCount;
-            _initialRangeSize =
-                initialTransferLength ??
-                ((long?)transferOptions.MaximumTransferLength) ??
-                Constants.DefaultBufferSize;
-            _rangeSize = Math.Min(
-                Constants.Blob.Block.MaxDownloadBytes,
-                transferOptions.MaximumTransferLength ?? Constants.DefaultBufferSize);
+
+            if (transferOptions.MaximumConcurrency.HasValue)
+            {
+                Debug.Assert(transferOptions.MaximumConcurrency > 0);
```

Comment:
We should just throw if we're not going to coerce this to something that will work correctly.

Reply:
Removed the debugs, now setting all values to defaults from ```Constants``` if user provides a negative number for ```MaximumConcurrency```, ```InitalTransferLength```, and ```MaximumTransferLength```.